src/pydvl/utils/types.py

```diff
@@ -64,3 +74,23 @@ def wrapper(*args, **kwargs):
         return fun(*args, **kwargs)
 
     return wrapper
+
+
+Seed = Optional[Union[int, np.random.SeedSequence]]
+SeedOrGenerator = Union[Seed, np.random.Generator]
```

Comment:
Problem is `numpy<1.25`. In upper versions Generator and SeedSequence share the same spawn method. If we support Python 3.8 we can't update numpy and this need to stick to that pattern, where two cases appear:

1. Functions which parallelize need a SeedSequence. There is no beautiful way to go from Generator back to SeedSequence. 
2. Functions which aren't parallelized, can also take a Generator instance. If we want to reduce it we have to stick here with SeedSequence, because case 1 can't be set to Generator.

Reply:
@mdbenito

Reply:
That is a good idea. I took over the later proposed function ensure_seed_seq below.